helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -874,13 +873,12 @@ private void unregisterAllEventMonitors() {
   }
 
   private void registerPerInstanceResources(Collection<PerInstanceResourceMonitor> monitors)
-      throws MalformedObjectNameException {
+      throws JMException {
     synchronized (_perInstanceResourceMonitorMap) {
       for (PerInstanceResourceMonitor monitor : monitors) {
         String instanceName = monitor.getInstanceName();
         String resourceName = monitor.getResourceName();
-        String beanName = getPerInstanceResourceBeanName(instanceName, resourceName);
-        register(monitor, getObjectName(beanName));
```

Comment:
Is it okay that this method (`private void register(Object bean, ObjectName name)`) is still used in other places in the code?

Reply:
Yeah, as long as we changed the register of workflow monitor and perinstanceresource monitor to use the dynamic register, we shouldn't be concerned about other places in this monitor. Later we may still need to change them though if we would like to change corresponding monitors.

Reply:
I see, thanks for clarification, but what about in other classes? My main concern is its usage in `public void active()` method, which in turn is used in `private void enableClusterStatusMonitor(boolean enable)` method in `GenericHelixController`. Although since the tests are passing it should be fine, but I was just curious.

Reply:
I see. So this register is only registering cluster status monitor, which we haven't changed it yet. And inside cluster status monitor, other monitors got registered individually with different methods. Once we start to change cluster status monitor, we may need to handle its register from its calling class.

Reply:
Got it, thanks.